src/selectors/Entities/name.js

```diff
@@ -101,21 +105,22 @@ export const selectVaccinePatientHistory = patient => {
         1, // Currently not possible to dispense more than 1 dose
       totalQuantity: 1,
       entryDate,
-      confirmDate:
-        (vaccinationData.extra?.prescription?.customData &&
-          JSON.parse(vaccinationData.extra?.prescription?.customData).dateOfVaccination &&
-          new Date(
-            convertMobileDateToISO(
-              JSON.parse(vaccinationData.extra?.prescription?.customData).dateOfVaccination
-            )
-          )) ??
-        'N/A',
+      confirmDate: hasDateOfVaccinationInSchema
```

Comment:
this is getting hard to follow!
I'd create

```
const getDateOfVaccination = (supplementalDataSchema, vaccinationData) => {
  const noDateOfVaccinationInSchema =
    !supplementalDataSchema?.jsonSchema?.properties?.dateOfVaccination;

  if (noDateOfVaccinationInSchema) return null;
  if (!vaccinationData.extra?.prescription?.customData) return 'N/A';

  const { dateOfVaccination } = JSON.parse(vaccinationData.extra?.prescription?.customData);
  if (!dateOfVaccination) return 'N/A';

  return new Date(convertVaccinationEntryToISOString(dateOfVaccination));
};
```

and then 

```
      confirmDate:
        getDateOfVaccination(supplementalDataSchema, vaccinationData) ?? new Date(entryDate),
```

Reply:
Haha - soz yes I re-read it and realised how confusing that was. Will adjust!